src/main/java/com/dsrm/dsrmbackend/services/impl/UserServiceImpl.java

```diff
@@ -35,4 +36,20 @@ public Page<User> getUsers(Pageable pageable){
         return userRepo.findAll(pageable);
     }
 
+    @Override
+    public Optional<User> updateUser(UserRolesOnlyDTO userRolesOnlyDTO, Long id) {
+        final Optional<User> user = userRepo.findById(id);
+        user.ifPresent(userEntity -> updateUserRoles(userRolesOnlyDTO, userEntity));
+        return user;
+    }
+
+    @Override
+    public void updateUserRoles(UserRolesOnlyDTO userRolesOnlyDTO, User user) {
+        if (userRolesOnlyDTO == null || userRolesOnlyDTO.getRoles() == null)
+            return;
+        User userRole = userMapper.toUserRoles(userRolesOnlyDTO);
```

Comment:
we don't want to create existing roles entities that. It is very error prone. Right way is
```
        Set<Role> userRoles = userRolesOnlyDTO.getRoles().stream().map(roleRepo::getReferenceById)
                        .collect(Collectors.toSet());
        user.setRoles(userRoles);
``` because we want to use existing entities managed by hibernate. Hint about getReferenceById method https://vladmihalcea.com/manytoone-jpa-hibernate/